src/server/game/AI/CreatureAI.cpp

```diff
@@ -134,7 +134,10 @@ void CreatureAI::MoveInLineOfSight(Unit* who)
         return;
 
     if (me->HasReactState(REACT_AGGRESSIVE) && me->CanStartAttack(who, false))
+    {
+        me->AddThreat(who, 0.0f); // ensure our initial target is the first thing added to threat list so we don't randomly switch off if DoZoneInCombat is called during the EnterCombat hook
```

Comment:
valid point about returning false - but it cannot be in AttackStart because that can be overridden by scripts (and you really dont want to care about such detail in every script yourself if you do that)